packages/api/src/external/carequality/command/cq-directory/create-or-update-cq-organization.ts

```diff
@@ -0,0 +1,51 @@
+import { CarequalityManagementAPI } from "@metriport/carequality-sdk";
+import { makeCarequalityManagementAPI } from "../../api";
+import { CQOrganization } from "../../organization";
+import { CQOrgDetails } from "../../shared";
+
+const cq = makeCarequalityManagementAPI();
+
+export async function createOrUpdateCQOrganization(orgDetails: CQOrgDetails): Promise<void> {
+  if (!cq) throw new Error("Carequality API not initialized");
+  const org = CQOrganization.fromDetails(orgDetails);
+  const cqOrgString = org.toXmlString();
+  const orgExists = await doesCQOrganizationExist(org.oid);
+  if (orgExists) {
+    await updateCQOrganization(cq, cqOrgString, org.oid);
+  } else {
+    await registerOrganization(cq, cqOrgString);
+  }
+}
+
+async function doesCQOrganizationExist(oid: string): Promise<boolean> {
+  if (!cq) throw new Error("Carequality API not initialized");
```

Comment:
NABD, but if we have the local pattern to pass the API to the other 2 functions, could do the same here and avoid this check